### DISCUSSION TITLE
docker push encounter unauthorized problem

### DISCUSSION THREAD

**Discussion by shangmin2009** [goharbor/harbor#21091] [Q&A]
I started harbor v2.11.1，and created a user shangmin in the website, and I can login with the user successfully.
And I created a project backend, and make sure the user shangmin is the maintainer of the project.
But when I push image to docker, it always encounters a problem:
docker push harbor.xiaoyubot.com/backend/xiaoyu-api:latest
unauthorized: unauthorized to access repository: backend/xiaoyu-api, action: push: unauthorized to access repository: backend/xiaoyu-api, action: push

Can someone give me some hints to solve the problem?

**Comment by Vad1mo** [comment]
Make sure you have the correct [external-url ](https://github.com/container-registry/harbor-helm?tab=readme-ov-file#configure-the-external-url ) or env `EXT_ENDPOINT`  with protocol://domain[:port] and also use that to log in and push.

**Reply by shangmin2009** [reply]
Thanks. I solved the problem by checking the conf of nginx running in the host machine and harbor.yml，and also set external_url in yml
